### PR TITLE
IsValidLiteralValue small optimization

### PR DESCRIPTION
**Pull request by sungam3r**

### CODE REVIEW

File: src/GraphQL/GraphQLExtensions.cs [hunk 1]
```diff
@@ -153,7 +153,9 @@ public static IGraphType BuildNamedType(this Type type, Func<Type, IGraphType> r
                        $"Expected non-null value, {nameof(resolve)} delegate return null for \"${type}\"");
         }
 
-        public static IEnumerable<string> IsValidLiteralValue(this IGraphType type, IValue valueAst, ISchema schema)
+        private static readonly string[] _foundNull = new[] { "Expected non-null value, found null" };
```
Comment (sungam3r) [inline src/GraphQL/GraphQLExtensions.cs:156]: I do not think that the ability to change the value of an array element should be considered a threat.

File: src/GraphQL/GraphQLExtensions.cs [hunk 2]
```diff
@@ -237,7 +239,7 @@ public static IEnumerable<string> IsValidLiteralValue(this IGraphType type, IVal
                     errors.AddRange(result.Select(err => $"In field \"{field.Name}\": {err}"));
                 }
 
-                return errors;
+                return errors.ToArray();
```
Comment (sungam3r) [inline src/GraphQL/GraphQLExtensions.cs:242]: In theory, this adds allocations, but in practice there should still be less of them due to the removal of `ToList` from other places 😕

**Review by Shane32 — APPROVED** [review]
Looks great!

**Comment by sungam3r** [comment]
but formatting...

**Pull request merged**